plugins/scenario/hu.bme.mit.gamma.scenario.model/src/hu/bme/mit/gamma/scenario/model/reduction/SimpleScenarioGenerator.java

```diff
@@ -74,9 +77,10 @@ public SimpleScenarioGenerator(ScenarioDefinition base, boolean transformLoopFra
 	private InteractionFragment previousFragment = null;
 	GammaEcoreUtil ecoreUtil = GammaEcoreUtil.INSTANCE;
 
-	public ScenarioDefinition execute() {
+	public ScenarioDeclaration execute() {
 		factory = ScenarioModelFactory.eINSTANCE;
-		simple = factory.createScenarioDefinition();
+		expressionFactory = ExpressionModelFactory.eINSTANCE;
```

Comment:
Extract the initialization of the factory object attributes to the declaration of the attributes (replace the null initial value with them).